indico/htdocs/js/indico/modules/abstracts.js

```diff
@@ -41,20 +42,40 @@
     global.setupAbstractPage = function setupAbstractPage() {
         $('body').on('declarative:success', '.js-delete-comment', function() {
             $(this).closest('.i-timeline-item').remove();
+        }).on('ajaxForm:show', '.js-edit-comment', function() {
+            $(this).closest('.abstract-comment-badges').hide();
+        }).on('ajaxForm:hide', '.js-edit-comment', function() {
+            $(this).closest('.abstract-comment-badges').show();
         }).on('focus', '.new-comment textarea', function() {
-            var $commentForm = $(this).closest('form');
+            var $box = $('#abstract-timeline-input');
+            var $commentForm = $box.find('form');
+            $box.find('.review-trigger').hide('blind', {direction: 'left'}, 'fast');
             $commentForm.find('.form-group').show('fast');
             $commentForm.removeClass('unfocused');
-        }).on('click', '.new-comment', function(evt) {
-            evt.stopPropagation();
-        });
-
-        $(window).on('click', function() {
-            var $commentForm = $('form.new-comment');
-            if (!$commentForm.find('textarea').val().trim().length) {
+            $commentForm.trigger('ajaxForm:externalShow');
+        }).on('click', '.new-comment .js-new-cancel', function(evt) {
+            evt.preventDefault();
+            var $this = $(this);
+            var $commentForm = $this.closest('form');
+            var $box = $this.closest('#abstract-timeline-input');
+            var $reviewTrigger = $box.find('.review-trigger');
+            var deferred = $.Deferred();
+            $commentForm.trigger('ajaxForm:externalHide', [deferred]);
+            deferred.then(function() {
+                $commentForm[0].reset();
+                $commentForm.trigger('change');
                 $commentForm.addClass('unfocused');
-                $commentForm.find('.form-group ~ .form-group').hide('fast');
-            }
+                $commentForm.find('.form-group ~ .form-group').hide('fast', function() {
+                    $reviewTrigger.show('blind', {direction: 'left'}, 'fast');
+                });
+            });
+        }).on('click', '.js-new-edit-review', function() {
+            var reviewId = $(this).data('reviewId');
+            var $reviewBox = $('#abstract-review-{0}'.format(reviewId));
+            $reviewBox.find('.js-edit-review').click();
```

Comment:
I usually prefer `.trigger('click')` since it's more explicit (`.click()` is one of those sligthly ugly trigger-event-or-bind-event-depending-on-the-arguments functions)